collect_app/src/androidTest/java/org/odk/collect/android/formentry/DeletingRepeatGroupsTest.java

```diff
@@ -0,0 +1,282 @@
+package org.odk.collect.android.formentry;
+
+import android.Manifest;
+
+import androidx.test.espresso.intent.rule.IntentsTestRule;
+import androidx.test.rule.GrantPermissionRule;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.odk.collect.android.R;
+import org.odk.collect.android.activities.FormEntryActivity;
+import org.odk.collect.android.espressoutils.FormEntry;
+import org.odk.collect.android.support.CopyFormRule;
+import org.odk.collect.android.support.ResetStateRule;
+import org.odk.collect.android.support.matchers.RecyclerViewMatcher;
+import org.odk.collect.android.test.FormLoadingUtils;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.action.ViewActions.longClick;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+
+public class DeletingRepeatGroupsTest {
+    private static final String TEST_FORM = "repeat_groups.xml";
+
+    @Rule
+    public IntentsTestRule<FormEntryActivity> activityTestRule = FormLoadingUtils.getFormActivityTestRuleFor(TEST_FORM);
+
+    @Rule
+    public RuleChain copyFormChain = RuleChain
+            .outerRule(GrantPermissionRule.grant(
+                    Manifest.permission.READ_EXTERNAL_STORAGE,
+                    Manifest.permission.WRITE_EXTERNAL_STORAGE
+                    )
+            )
+            .around(new ResetStateRule())
+            .around(new CopyFormRule(TEST_FORM, null));
+
+    @Test
+    public void testDeletingRepeatGroupsInForm() {
```

Comment:
I see two high-level issues with these tests:
- they're going to take a long time to run because there's a lot of swiping back and forth
- if one fails, it's going to be hard to know what that means since the cases aren't identified

What are the cases that need to be verified? Seems to me that they are deleting the first repeat, deleting the last repeat and deleting some repeat in the middle. What is the fourth case you have? Is it necessary? 

Why are all of the repeats after the deleted ones verified? Would perhaps just checking the one before and the one after be enough?

Why are there two questions in each repeat? I think it's to visually differentiate the field list and non-field list case but does that actually matter for the test? That is, would the field list test fail before the fix even if there was just a single question in each repeat? I think so and in that case, it'd be simpler to have a single question in each.

Since populating the repeats is not part of the test, could you set the form definition to include them? I mean having something like the following in the form definition `instance` block:
```
<repeatGroup>
  <text>foo</text>
</repeatGroup>
<repeatGroup>
  <text>bar</text>
</repeatGroup>
...
<repeatGroupFieldList>
  <number>12</number>
</repeatGroupFieldList>
<repeatGroupFieldList>
  <number>24</number>
</repeatGroupFieldList>
```

That way you can jump directly to the repeat to delete and avoid a lot of the swiping.

Then it also makes it easier to have independent cases like `requestingDeletionOfFirstRepeat_deletesFirstRepeat` and `requestingDeletionOfMiddleRepeat_deletesMiddleRepeat`.

Reply:
ok I simplified those tests and divided them into more but smaller.